polygeist/tools/cgeist/Test/Verification/sycl/sycl_use_host_module.cpp

```diff
@@ -0,0 +1,12 @@
+// RUN: clang++ -fsycl -fsycl-device-only -Xcgeist -sycl-use-host-module=%S/host_module.mlir -O0 -w -emit-mlir -o - %s | FileCheck %s
+
+#include <sycl/sycl.hpp>
+
+// CHECK: module attributes
+// CHECK:   gpu.module @device_functions
+// CHECK:   memref.global constant @c : memref<i32, 1> {alignment = 4 : i64}
```

Comment:
So the idea here is to link in the code in `host_module.mlir` into the current module. The function definition for `do_nothing` should be in the top level module. The way the test is written makes me think that the function might be inside the gpu module. Can you modify the test to show that cannot happen pls ?

Reply:
I added a small FileCheck trick for that